Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/DataSynchronizer.swift

```diff
@@ -400,563 +405,688 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
-                            )
-                            continue
                         }
-                    } else {
-                        nextVersion = nil
                     }
 
-                    logger.i(
-                        "t='\(logicalT)': syncLocalToRemote ns=\(nsConfig.namespace) documentId=\(docConfig.documentId) conflict=\(isConflicted)")
-
-                    if !isConflicted {
-                        // iv. If no conflict has occurred, move on to the remote to local sync routine.
-
-                        // since we strip version information from documents before setting pending writes, we
-                        // don't have to worry about a stale document version in the event here.
-                        let committedEvent = docConfig.uncommittedChangeEvent!
-                        self.emitEvent(documentId: docConfig.documentId.value, event: ChangeEvent<Document>(
-                            id: committedEvent.id,
-                            operationType: committedEvent.operationType,
-                            fullDocument: committedEvent.fullDocument,
-                            ns: committedEvent.ns,
-                            documentKey: committedEvent.documentKey,
-                            updateDescription: committedEvent.updateDescription,
-                            hasUncommittedWrites: false))
-
-                        try docConfig.setPendingWritesComplete(atVersion: nextVersion)
-                    } else {
-                        // v. Otherwise, invoke the collection-level conflict handler with the local change
-                        // event and the remote change event (synthesized by doing a lookup of the document or
-                        // sourced from the listener)
-                        let remoteChangeEvent: ChangeEvent<Document>
-                        if !remoteDocumentFetched {
-                            remoteChangeEvent =
-                                try synthesizedRemoteChangeEvent(in: remoteColl, with: docConfig.documentId.value)
-                        } else {
-                            remoteChangeEvent =
-                                synthesizedRemoteChangeEvent(
-                                    for: MongoNamespace.init(databaseName: remoteColl.databaseName, collectionName: remoteColl.name),
-                                    with: docConfig.documentId.value,
-                                    for: remoteDocument)
+                    if setPendingWritesComplete, let uncommittedChangeEvent = docConfig.uncommittedChangeEvent {
+                        if let uncommittedDocument = uncommittedChangeEvent.fullDocument {
+                            docConfig.setPendingWritesComplete(atHash: HashUtils.hash(doc:
+                                DataSynchronizer.sanitizeDocument(uncommittedDocument)), atVersion: nextVersion)
+
+                            let opType: 
```

Comment:
[style] the `DataSynchronizerError!` implicitly unwrapped var seems like it is often `nil` (i.e. the non-error case), so it should just be a `DataSynchronizerError?`. Implicitly unwrapped optionals should only be used in cases where access is guaranteed to be safe.

Reply:
ok